service/src/test/java/uk/nhs/adaptors/gp2gp/utils/DateFormatUtilTest.java

```diff
@@ -64,6 +67,15 @@ public void When_FormattingDateTimeTypeToText_Expect_Hl7InUkZone(String input, S
         assertThat(actual).isEqualTo(expected);
     }
 
+    @ParameterizedTest
+    @CsvSource({"20190328103005,20190328103005", "1973,19730101000000", "200808,20080801000000", "20190728,20190728000000"})
+    public void When_FormattingDateTimeTypeFromHl7FormatTextBackToDate_Expect_Hl7InUkZone(String firstParsedDate,
+        String expectedConvertedBackDateTimeTpeString) {
```

Comment:
Should this be 'type'?
```suggestion
        String expectedConvertedBackDateTimeTypeString) {
```